### PR TITLE
Adding code coverage CI

### PR DESCRIPTION
**Pull request by kevinlewi**
Adding a new CI pipeline for measuring code coverage. Followed this guide here: https://blog.rng0.io/how-to-do-code-coverage-in-rust

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,10 @@
 .DS_Store
 .vs
 
+debug/
+target/
```
Comment (slawlor) [inline .gitignore:5]: This won't be excluded already with **/target?

File: xtask/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[package]
```
Comment (slawlor) [inline xtask/Cargo.toml:1]: This looks like a binary, do we need a `default-run` statement?

**Review by slawlor — APPROVED** [review]
Looks good, just a couple comments for running the coverage but this is helpful thanks!

File: .gitignore (hunk 1 above)
Comment (kevinlewi) [reply .gitignore:5]: Ah, thanks!

**Pull request merged**